### ISSUE TITLE
Show dependency information on ZenHub board

### ISSUE THREAD

**Issue by MFDW** [ZenHubIO/support#1184]
## user story
As a ZenHub using Product Owner
When I have created a dependency using ZenHub
And am reviewing the ZenHub project board
I want to be able to see the dependency information on the board
So that I can be reminded of blocking tasks
And plan work effectively by doing the most appropriate work first

Info shown on the issue
![image](https://user-images.githubusercontent.com/11423313/28209502-a210ca56-688a-11e7-8cd6-621b76c15173.png)

Does not show on the zenhub board
![image](https://user-images.githubusercontent.com/11423313/28209466-842a9b52-688a-11e7-90d3-26ab43efafed.png)

## suggestions/ideas
- something like how `connects to #12345` shows PRs against an issue, possibly showing blockers above the issue as a one-liner in a vertical list with an expandable option to see full name of each blocker. Blocking issues could show the issues being blocked by it beneath the issue.
- at the minimum I think a label like the one showing on the issue would be good `blocked` and `blocker` possibly with a count of the number of issues blocking this issue or issues being blocked by this issue. Should also have links to the blocking/blocker issues from the board view as well.

**Comment by tiabc** [comment]
+1
In status.im we also use ZenHub and we find this product generally amazing. And right today we've decided to introduce `blocker` and `blocked` labels just to make it more apparent which issues are blocked on board. That'd be incredibly helpful to have dependencies information at hand.

**Comment by pguedes** [comment]
+1 this would be very helpful

**Comment by pablanopete** [comment]
+1

**Comment by azenMatt** [comment]
Hi everyone, we are working on some ideas for this and will have some designs to share soon!

**Comment by azenMatt** [comment]
Hi everyone, 

We are starting to work on this at the moment, based on the following ideas. 

The main change will be to include a new notifier on the issue card, as well as the ability to filter by both upstream and downstream dependancies 

As the bottom half of the issue card is already quite busy, dependancy information will be added to the top right hand corner. Showing an icon for the three states:

- Blocked
- Blocking 
- Both blocked and blocking. 

When selected, this icon will provide more context, like so:

![image](https://user-images.githubusercontent.com/7584089/30233706-1ed69fbe-94ac-11e7-92ed-92a1cced745e.png)

If the open to filter is selected, the user will only see those issues that make up the dependancy tree on the board:

![image](https://user-images.githubusercontent.com/7584089/30233723-35d5ca5a-94ac-11e7-98fd-db1d62702bc7.png)


![image](https://user-images.githubusercontent.com/7584089/30233748-5de3b2c8-94ac-11e7-94b5-6019d61eacd3.gif)

In the future, we would like to improve this information to also show the tree structure as a visualized graph. This can be worked out from the information in the board in simple cases like the examples above, but is soon going to get complex. Which is where we think the tree structure would add additional value

This is out of scope for the first changes here, but we are exploring options of how that could look for a future release. 


Thoughts on the ideas above? Will this make working with dependancies more useful for your teams?

**Comment by davidklaw** [comment]
Very much like this idea. It's what I had envisioned when looking to see if this was already in the hopper. 👍

**Comment by MFDW** [comment]
Looks great :+1:  nice work

Thoughts from me
- triangle & square makes sense but perhaps difference is not immediately obvious. Some kind of colour would help. Either colour the border of the shape or fill it. It shouldn't pop on the board view but when you look you should be immediately able to spot the difference without 'analysing'. 
I think perhaps using a filled grey square with a white exclamation mark might do the trick?
- Been thinking about this a bit. The scrum daily standup questions includes 'is anything blocking you?' and as such we, and I'm sure many teams, use a 'blocked' label to indicate work being blocked off a sprint. Using the same terminology may be problematic. We solve this previously by using 'waiting on dependent' but it's awkward phrasing. 
I don't have a solution but thought it was worth a mention..
- perhaps keep the 3 dots always visible? Add it alongside the icon rather than replacing it to show core functionality has been added to rather than changed.
Though making it change when hovering on the whole issue makes sense..

Seems like you've done a great job of not cluttering the board but making the information readily available at both a project and individual task level :+1:

**Comment by azenMatt** [comment]
Thank you for the feedback @MFDW 

I agree with all your thoughts here 😄 



> triangle & square makes sense but perhaps difference is not immediately obvious. Some kind of colour would help. Either colour the border of the shape or fill it. It shouldn't pop on the board view but when you look you should be immediately able to spot the difference without 'analysing'.
> I think perhaps using a filled grey square with a white exclamation mark might do the trick?

We tried quite a few different mock ups here. We all really wanted one with colour to be the best! but they never quite felt right. Our hope is that once a team has got used to the symbols they will be much clearer. I think we can check in on this a few weeks post launch

>Been thinking about this a bit. The scrum daily standup questions includes 'is anything blocking you?' and as such we, and I'm sure many teams, use a 'blocked' label to indicate work being blocked off a sprint. Using the same terminology may be problematic. We solve this previously by using 'waiting on dependent' but it's awkward phrasing.
I don't have a solution but thought it was worth a mention..

Something we have been thinking about along these lines, is custom dependancies. This might help here by allowing someone to say they are blocked by something which isn't an issue? 

> perhaps keep the 3 dots always visible? Add it alongside the icon rather than replacing it to show core functionality has been added to rather than changed.

We moved over to the three dots last week, these show on hover - which is a bit more noticeable than the previous 'cog' icon. Having them appear all the time was a bit much we felt. 

Looking forward to getting this into your hands for feedback sometime next week ( 🤞 )

**Comment by kendramoroz** [comment]
Today's the day! View Dependencies on the Board and filtering by an Issue's Dependency tree is live 🚀 

[Take a look through the full guide here for more information](https://www.zenhub.com/guides/issue-dependencies#dependencies-on-the-board) and be sure to let us know what you think! 

![image](https://user-images.githubusercontent.com/11036230/31025361-430857d2-a4f7-11e7-83ad-18bb74342fea.png)

![image](https://user-images.githubusercontent.com/11036230/31025356-3d0e7e38-a4f7-11e7-97ed-6a0dadde5433.png)

**Comment by MFDW** [comment]
Nice one :+1:

**Comment by MFDW** [comment]
Nice one :+1: 

![image](https://user-images.githubusercontent.com/11423313/31068153-978329fa-a74e-11e7-9a3b-a5e8fc50bc70.png)

If there are multiple levels in the three (i.e grandparents and grandchildren) will they show too?
Wasn't something I'd considered..

**Comment by kendramoroz** [comment]
@MFDW Hey Dave, you got it! All children, grandchildren, etc. that are in the repos that you are filtering by on your Multi-repo Board (if applicable) will be shown 😊 .

**Comment by mcupak** [comment]
Very nice! It would be great if we could show all issues blocked by something, e.g. similarly to filtering by a label. Is this possible?

**Comment by kendramoroz** [comment]
@mcupak great suggestion! I'll issue this up internally for consideration as we continue to enhance Dependencies 🎉  By default, the Board will highlight through icons all blocked Issues and their relative relationship to other Issues, but there's no direct option to hide non-Dependency oriented Issues.

**Comment by shanemileham** [comment]
@kendramoroz On the flip side, would it also be possible to only show tasks without dependencies? Possibly color-coded by number of dependencies. That would be extremely helpful to know how to allocate current resources to available tasks and free up dependent ones.

**Comment by kendramoroz** [comment]
@shanemileham thanks for the additional feedback here! Not currently an option for dependency filtering, but I've created an issue here that you can follow on any enhancements we make to this area: https://github.com/ZenHubIO/support/issues/1278 🚀

**Comment by ryanhiebert** [comment]
A graph view was mentioned briefly on this issue. Is there another issue that  I can follow development on that idea?
